spring-cloud-contract-wiremock/src/main/java/org/springframework/cloud/contract/wiremock/restdocs/WireMockSnippet.java

```diff
@@ -90,8 +79,17 @@ private void extractMatchers(Operation operation) {
 				.get("contract.jsonPaths");
 		this.jsonPaths = jsonPaths;
 		this.contentType = (MediaType) operation.getAttributes().get("contract.contentType");
+		if (this.contentType == null){
+			this.hasJsonBodyRequestToMatch = hasJsonContentType(operation);
+		}
 	}
 
+	private boolean hasJsonContentType(Operation operation) {
+		return operation.getRequest().getHeaders().getContentType() != null 
```

Comment:
We can extract this to a local variable `operation.getRequest().getHeaders().getContentType()` ?